src/main/scala/org/apache/spark/sql/delta/sources/DeltaSource.scala

```diff
@@ -285,6 +292,8 @@ case class DeltaSource(
     val changes = if (start.isEmpty) {
       if (endOffset.isStartingVersion) {
```

Comment:
```scala
getStartingVersion match {
  case Some(v) => getChanges(v, -1L, isStartingVersion = false)
  case _ =>
     if (endOffset.isStartingVersion) {
       getChanges(endOffset.reservoirVersion, -1L, isStartingVersion = true)
     } else {
        assert(endOffset.reservoirVersion > 0, s"invalid reservoirVersion in endOffset: $endOffset")
        // Load from snapshot `endOffset.reservoirVersion - 1L` so that `index` in `endOffset`
        // is still valid.
        getChanges(endOffset.reservoirVersion - 1L, -1L, isStartingVersion = true)
      }
}

```

Reply:
I think we need to use `start` and `start.get.isStartingVersion` passed into `getBatch`.

Reply:
this is the case for `start` being empty though